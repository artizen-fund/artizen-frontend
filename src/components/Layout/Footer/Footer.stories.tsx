import Footer from './'

export default {
  title: 'layout/Footer',
  component: Footer,
}

export const FooterComponent = () => {
  return <Footer />
}
