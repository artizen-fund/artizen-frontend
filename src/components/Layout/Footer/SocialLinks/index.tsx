import styled from 'styled-components'
import Link from 'next/link'
import { breakpoint, palette, typography } from '@theme'
import { rgba } from '@lib'

const SocialLinks = () => {
  return (
    <Wrapper>
      <SocialLink href="https://twitter.com" icon="twitter">
        Twitter
      </SocialLink>
      <SocialLink href="https://discord.com" icon="discord">
        Discord
      </SocialLink>
      <SocialLink href="https://github.com" icon="github">
        Github
      </SocialLink>
      <SocialLink href="https://telegram.com" icon="telegram">
        Telegram
      </SocialLink>
    </Wrapper>
  )
}
const Wrapper = styled.div`
  display: flex;
  flex-direction: row;
  gap: 8px;
  @media only screen and (min-width: ${breakpoint.tablet}px) {
    gap: 12px;
  }
  @media only screen and (min-width: ${breakpoint.desktop}px) {
    gap: 16px;
  }
`

const SocialLink = styled.a<{ icon: string }>`
  display: block;
  width: 56px;
  height: 56px;
  @media only screen and (min-width: ${breakpoint.tablet}px) {
    width: 64px;
    height: 64px;
  }
  @media only screen and (min-width: ${breakpoint.desktop}px) {
    width: 72px;
    height: 72px;
  }

  text-indent: -1000px;
  overflow: hidden;
  border-radius: 9999px;
  background-color: ${rgba(palette.white)};
  background-repeat: no-repeat;
  background-position: center center;
  background-image: url('/socialIcons/${props => props.icon}.svg');
`

export default SocialLinks
