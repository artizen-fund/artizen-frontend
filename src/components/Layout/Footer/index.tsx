import { useState } from 'react'
import styled from 'styled-components'
import { breakpoint, palette, typography } from '@theme'
import { rgba } from '@lib'
import { Button, PagePadding } from '@components'
import SocialLinks from './SocialLinks'

const Footer = () => {
  return (
    <>
      <PagePadding black>
        <Wrapper>
          <Rule gridSpace="rule1" />
          <BigLinks>
            <li>Mission</li>
            <li>Leaderboard</li>
            <li>How it Works</li>
            <li>Community</li>
          </BigLinks>
          <SmallLinks>
            <li>Contact</li>
            <li>Careers</li>
            <li>Press</li>
          </SmallLinks>
          <Rule gridSpace="rule2" />
          <CTA>
            <Copy>Join the mission to fund public goods.</Copy>
            <Buttons>
              <Button onClick={() => alert('coming soon')} outline size="l1">
                Donate
              </Button>
              <Button onClick={() => alert('coming soon')} outline size="l1">
                Sign In
              </Button>
            </Buttons>
          </CTA>
          <Rule gridSpace="rule3" />
          <SocialLinks />
        </Wrapper>
      </PagePadding>
      <StyledPagePadding>
        <Credits>
          <div>Open source platform made with ❤️ by a globally distributed team</div>
          <ul>
            <li>Privacy Policy</li>
            <li>Terms &amp; Conditions</li>
          </ul>
        </Credits>
      </StyledPagePadding>
    </>
  )
}

const Wrapper = styled.footer`
  display: grid;
  grid-gap: 18px;
  grid-template-areas:
    'rule1 rule1'
    'bigLinks smallLinks'
    'rule2 rule2'
    'cta cta'
    'rule3 rule3'
    'social social';
  grid-template-columns: repeat(2, 1fr);
  @media only screen and (min-width: ${breakpoint.laptop}px) {
    grid-gap: 20px 40px;
    grid-template-areas:
      'rule1 rule1 rule2'
      'bigLinks smallLinks cta'
      'rule3 rule3 rule3'
      'social social social';
    grid-template-columns: 2fr 1fr 3fr;
    /* todo: this grid template is not getting us equal columns */
  }
  @media only screen and (min-width: ${breakpoint.desktop}px) {
    grid-gap: 30px 65px;
  }
  color: ${rgba(palette.white)};
`

const Rule = styled.hr<{ gridSpace: string }>`
  grid-area: ${props => props.gridSpace};
  display: block;
  appearance: none;
  width: 100%;
  height: 1px;
  background-color: ${rgba(palette.barracuda)};
  border: 0px;
  outline: 0px;
`

const BigLinks = styled.ul`
  grid-area: bigLinks;
  ${typography.title.l2}
`

const SmallLinks = styled.ul`
  grid-area: smallLinks;
  ${typography.label.l0}
`

const CTA = styled.div`
  grid-area: cta;
`

const Copy = styled.div`
  ${typography.title.l2}
`

const Buttons = styled.div`
  @media only screen and (min-width: ${breakpoint.laptop}px) {
    display: flex;
    flex-direction: row;
    gap: 10px;
  }
`

const Credits = styled.footer`
  text-align: center;
  ${typography.label.l2}
  color: ${rgba(palette.barracuda)};
  ul {
    display: flex;
    justify-content: space-around;
    li {
      display: block;
    }
  }
  @media only screen and (min-width: ${breakpoint.laptop}px) {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
  }
`

const StyledPagePadding = styled(props => <PagePadding {...props} />)`
  padding: 15px 0;
  background-color: ${rgba(palette.white)};
  @media (prefers-color-scheme: dark) {
    background-color: ${rgba(palette.black)};
    color: ${rgba(palette.white)};
  }
`

export default Footer
