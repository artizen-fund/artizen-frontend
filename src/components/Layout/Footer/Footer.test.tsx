import '@testing-library/jest-dom/extend-expect'
import { render } from '@testing-library/react'
import Footer from './'

describe('Footer', () => {
  it('renders Footer unchanged', () => {
    const { container } = render(<Footer />)
    expect(container).toMatchSnapshot()
  })
})
