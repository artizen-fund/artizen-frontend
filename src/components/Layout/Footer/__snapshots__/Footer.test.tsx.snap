// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders Footer unchanged 1`] = `
<div>
  <div
    class="sc-b3ed1e8f-0 SyFSF"
  >
    <footer
      class="sc-77186b7a-0 deZMJj"
    >
      <hr
        class="sc-77186b7a-1 kvWYbx"
      />
      <ul
        class="sc-77186b7a-2 iWhbev"
      >
        <li>
          Mission
        </li>
        <li>
          Leaderboard
        </li>
        <li>
          How it Works
        </li>
        <li>
          Community
        </li>
      </ul>
      <ul
        class="sc-77186b7a-3 fSEVLL"
      >
        <li>
          Contact
        </li>
        <li>
          Careers
        </li>
        <li>
          Press
        </li>
      </ul>
      <hr
        class="sc-77186b7a-1 fZsNBC"
      />
      <div
        class="sc-77186b7a-4 ivFIpq"
      >
        <div
          class="sc-77186b7a-5 hUUstO"
        >
          Join the mission to fund public goods.
        </div>
        <div
          class="sc-77186b7a-6 lfnkI"
        >
          <button
            class="sc-a72af660-2 bDNfBj"
          >
            Donate
          </button>
          <button
            class="sc-a72af660-2 bDNfBj"
          >
            Sign In
          </button>
        </div>
      </div>
      <hr
        class="sc-77186b7a-1 ehgKyj"
      />
      <div
        class="sc-cbdb4d91-0 ehmDoo"
      >
        <a
          class="sc-cbdb4d91-1 QVaUY"
          href="https://twitter.com"
        >
          Twitter
        </a>
        <a
          class="sc-cbdb4d91-1 eAzSqZ"
          href="https://discord.com"
        >
          Discord
        </a>
        <a
          class="sc-cbdb4d91-1 kDaYjE"
          href="https://github.com"
        >
          Github
        </a>
        <a
          class="sc-cbdb4d91-1 iqfkFQ"
          href="https://telegram.com"
        >
          Telegram
        </a>
      </div>
    </footer>
  </div>
  <div
    class="sc-b3ed1e8f-0 jbZhgc sc-77186b7a-8 eBOeez"
  >
    <footer
      class="sc-77186b7a-7 hRDkMk"
    >
      <div>
        Open source platform made with ❤️ by a globally distributed team
      </div>
      <ul>
        <li>
          Privacy Policy
        </li>
        <li>
          Terms & Conditions
        </li>
      </ul>
    </footer>
  </div>
</div>
`;
